doc/tutorials/vuart_configuration.rst

```diff
@@ -214,3 +214,131 @@ started, as shown in the diagram below:
+        <target_vm_id>1</target_vm_id>
+        <target_uart_id>1</target_uart_id>
+      </communication_vuart>
+   </vm>
+
+   <vm id="1">
+      <legacy_vuart id="0">
+        <type>VUART_LEGACY_PIO</type>
+        <base>INVALID_COM_BASE</base>     /* vuart[0] is not used */
+      </legacy_vuart>
+      <legacy_vuart id="1">
+        <type>VUART_LEGACY_PIO</type>
+        <base>INVALID_COM_BASE</base>     /* vuart[1] is not used */
+      </legacy_vuart>
+      <console_vuart id="0">
+        <base>PCI_VUART</base>            /* PCI-vUART0 is console port */
+      </console_vuart>
+      <communication_vuart id="1">
+        <base>PCI_VUART</base>            /* PCI-vUART1 is commincation port, connect to vUART1 of VM0 */
+        <target_vm_id>0</target_vm_id>
+        <target_uart_id>1</target_uart_id>
+      </communication_vuart>
+   </vm>
+
+The ACRN vUART related xml fields:
+ - 'id' in <legacy_vuart>, value of vuart_idx, id=0 is for legacy vuart[0] configuration, id=1 is for vuart[1].
+ - 'type' in <legacy_vuart>, type is always VUART_LEGACY_PIO for legacy vUART.
+ - 'base' in <legacy_vuart>, is use the legacy vUART port, set COM1_BASE for vuart[0], set COM2_BASE for vuart[1].INVALID_COM_BASE means do not use the legacy vUART port.
+ - 'irq' in <legacy_vuart>, is use the legacy vUART port, set COM1_IRQ for vuart[0], set COM2_IRQ for vuart[1];
+ - 'id' in <console_vuart> and <communication_vuart>, vuart_idx for PCI-vUART
+ - 'base' in <console_vuart> and <communication_vuart>, PCI_VUART means use this PCI-vUART, it is a console. INVALID_PCI_BASE means do not use this PCI-VUART.
+ - 'target_vm_id' and 'target_uart_id', connection settings for this vUART port.
+
+Run ``make BOARD_FILE=$PWD/misc/acrn-config/xmls/board-xmls/<board>.xml SCENARIO_FILE=/home/taoyuhong/acrn/acrn-hypervisor/misc/acrn-config/xmls/config-xmls/<board>/<scenario.xml``
+to build ACRN with xml configuration file. Configuration tools will test your settings, check `vUART rules <https://projectacrn.github.io/latest/tutorials/vuart_configuration.html#index-of-vuart>`_ for compile issue. After compile, you can find ``./misc/vm_configs/scenarios/<scenario>/<board>/pci_dev.c`` has been changed by configuration tools, something like:
+
+.. code-block:: none
+
+   {
+      .emu_type = PCI_DEV_TYPE_HVEMUL,
+      .vbdf.bits = {.b = 0x00U, .d = 0x05U, .f = 0x00U},
+      .vdev_ops = &vmcs9900_ops,
+      .vbar_base[0] = 0x80003000,
+      .vbar_base[1] = 0x80004000,
+      .vuart_idx = 1,
+      .t_vuart.vm_id = 1U,
+      .t_vuart.vuart_id = 1U,
+    },
+
+That shows a PCI-vUART with vuart_idx=1, BDF 00:05.0, is PCI-vUART1 of this VM. We can use BDF to identify PCI-vUART in VM. Usually, legacy vuart[0] is ttyS0 in VM, and vuart[1] is ttyS1. So we hope PCI-vUART0 is ttyS0, PCI-VUART1 is ttyS1 ... PCI-vUART7 is ttyS7, but that is not true. If run ``dmesg | grep tty`` at VM shell, user may see:
```

Comment:
Not sure what we are trying to say here. Can you explain some more?

Reply:
Because user need to know which `"/dev/ttyS*" is the right file, should be read/wrote for communication. For example, user know VM0 has PCI-vUART1 connected to VM1 PCI-vUART1, VM0 has PCI-vUART2 connected to VM2 PCI-vUART1. When VM0 want to talk to VM1, and VM0 has ttyS1, ttyS2 ... ttyS5 . In VM0, user may want to know which ttyS* is PCI-vUART1. It is not ttyS1 when using ACRN kernel. 
HV configuration tool assigns the BDF value, mmio address of VM0 PCI-vUART1. By checking BDF and mmio, user knows the mapping between "/dev/ttyS*" and vUART*

Reply:
OK, thanks for the explanation, I think I understand this better now. We need to indicate that the use needs to retrieve which `ttyS*` is mapped to the device by inspecting `dmesg` to see which maps to the correct MMIO or BDF. Is that correct?